src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -24519,6 +25460,904 @@ static void M1()
+
+            var comp2 = CreateNullableCompilation(new[] { source, lib_cs, DisallowNullAttributeDefinition });
+            comp2.VerifyDiagnostics(expected);
+        }
+
+        [Fact]
+        public void DisallowNull_Indexer_OtherParameters_NullableInt()
+        {
+            var lib_cs =
+@"using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    public int? this[[DisallowNull] int? s] { set => throw null!; }
+}
+";
+
+            var lib = CreateNullableCompilation(new[] { DisallowNullAttributeDefinition, lib_cs });
+
+            var source = @"
+class D
+{
+    static void M1(int? i, int i2)
+    {
+        new C()[i] = i; // 1
+        new C()[i2] = i;
+    }
+}";
+            var expected = new[]
+            {
+                // (6,17): warning CS8607: A possible null value may not be passed to a target marked with the [DisallowNull] attribute
+                //         new C()[i] = i; // 1
+                Diagnostic(ErrorCode.WRN_DisallowNullAttributeForbidsMaybeNullAssignment, "i").WithLocation(6, 17)
+            };
+
+            var comp = CreateNullableCompilation(source, references: new[] { lib.EmitToImageReference() });
+            comp.VerifyDiagnostics(expected);
+
+            var comp2 = CreateNullableCompilation(new[] { source, lib_cs, DisallowNullAttributeDefinition });
+            comp2.VerifyDiagnostics(expected);
+        }
+
+        [Fact]
+        public void DisallowNull_Indexer_OtherParameters_OverridingGetter()
+        {
+            var lib_cs =
+@"using System.Diagnostics.CodeAnalysis;
+public class Base
+{
+    public virtual string this[[DisallowNull] string? s] { get => throw null!; }
+}
+public class C : Base
+{
+    public override string this[string? s] { get => throw null!; }
+}
+";
+
+            var lib = CreateNullableCompilation(new[] { DisallowNullAttributeDefinition, lib_cs });
+
+            var source = @"
+class D
+{
+    static void M1(string? s, string s2)
+    {
+        new C()[s].ToString();
+        new C()[s2].ToString();
+    }
+}";
+
+            var comp = CreateNullableCompilation(source, references: new[] { lib.EmitToImageReference() });
+            comp.VerifyDiagnostics();
+
+            var comp2 = CreateNullableCompilation(new[] { source, lib_cs, DisallowNullAttributeDefinition });
+            comp2.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void DisallowNull_Indexer_OtherParameters_OverridingGetterOnly()
+        {
+            var lib_cs =
+@"using System.Diagnostics.CodeAnalysis;
+public class Base
+{
+    public virtual string this[[DisallowNull] string? s] { get => throw null!; set => throw null!; }
+}
+public class C : Base
+{
+    public override string this[string? s] { set => throw null!; }
+}
+";
+
+            var lib = CreateNullableCompilation(new[] { DisallowNullAttributeDefinition, lib_cs });
+
+            var source = @"
+class D
+{
+    static void M1(string? s, string s2)
+    {
+        new C()[s].ToString(); // 1
```

Comment:
>new C()[s].ToString(); // 1 [](start = 8, length = 27)

Should there be a warning here?

Reply:
The number comment is stale. I'll remove

---
In reply to: [296036465](https://github.com/dotnet/roslyn/pull/36444#discussion_r296036465) [](ancestors = 296036465)